artifactory/commands/transferfiles/transfer.go

```diff
@@ -136,27 +136,32 @@ func (tdc *TransferFilesCommand) Run() (err error) {
 			tdc.initNewPhase(newPhase, srcUpService, repoSummary)
 			skip, err := newPhase.shouldSkipPhase()
 			if err != nil {
+				tdc.stopTransferAfterErrorOccurred()
```

Comment:
Notice that we're calling `tdc.stopTransferAfterErrorOccurred()` after when every error is caught here.
The code can call `tdc.stopTransferAfterErrorOccurred()` only once, if we implement the following refactoring:
1. Move the entire code of this loop into a new function.
2. After this function returns an error, call `tdc.stopTransferAfterErrorOccurred()` once.

Let's also rename the "phaseI" into "currentPhase".